src/server/routes/entity/entity.tsx

```diff
@@ -84,67 +84,68 @@ export function displayEntity(req: PassportRequest, res: $Response) {
-			bbid: resLocals.entity.bbid,
-			editorId: resLocals.user.id
-		})
-			.save(null, {method: 'insert'})
-			.then(() => achievement.processPageVisit(orm, resLocals.user.id))
-			.catch(
-				// error caused by duplicates we do not want in database
-				() => new Promise(resolve => resolve(false))
-			);
+		try {
+			await new EditorEntityVisits({
+				bbid: resLocals.entity.bbid,
+				editorId: resLocals.user.id
+			})
+				.save(null, {method: 'insert'});
+
+			editorEntityVisitPromise = await achievement.processPageVisit(orm, resLocals.user.id);
+		}
+		catch {
+			// error caused by duplicates we do not want in database
+			return false;
+		}
 	}
 	else {
 		editorEntityVisitPromise = new Promise(resolve => resolve(false));
 	}
 
-	let alertPromise = editorEntityVisitPromise.then((visitAlert) => {
-		let alertIds = [];
-		if (visitAlert.alert) {
-			alertIds = alertIds.concat(visitAlert.alert.split(',').map(
-				(id) => parseInt(id, 10)
-			));
-		}
-		if (_.isString(req.query.alert)) {
-			// $FlowFixMe
-			alertIds = alertIds.concat(req.query.alert.split(',').map(
-				(id) =>	parseInt(id, 10)
-			));
-		}
-		if (alertIds.length > 0) {
-			const promiseList = alertIds.map(
-				(achievementAlert) =>
-					new AchievementUnlock(
-						{id: achievementAlert}
-					)
-						.fetch({
-							require: true,
-							withRelated: 'achievement'
-						})
-						.then((unlock) => unlock.toJSON())
-						.then((unlock) => {
-							let unlockName;
-							if (req.user.id === unlock.editorId) {
-								unlockName = {
-									name: unlock.achievement.name
-								};
-							}
-							return unlockName;
-						})
-						.catch((err) => {
-							log.debug(err);
-						})
-			);
-			alertPromise = Promise.all(promiseList);
-		}
-		else {
-			alertPromise = new Promise(resolve => resolve(false));
-		}
-		return alertPromise;
-	});
+	let alertPromise;
+	const visitAlert = await editorEntityVisitPromise;
+	let alertIds = [];
+	if (visitAlert.alert) {
+		alertIds = alertIds.concat(visitAlert.alert.split(',').map(
+			(id) => parseInt(id, 10)
+		));
+	}
+	if (_.isString(req.query.alert)) {
+		// $FlowFixMe
+		alertIds = alertIds.concat(req.query.alert.split(',').map(
+			(id) =>	parseInt(id, 10)
+		));
+	}
+	if (alertIds.length > 0) {
+		const promiseList = alertIds.map(async (achievementAlert) => {
+			try {
+				const unlock = await new AchievementUnlock({id: achievementAlert}).fetch({
+					require: true,
+					withRelated: 'achievement'
+				});
+				const unlockJSON = unlock.toJSON();
 
-	return alertPromise.then((alert) => {
+				let unlockName;
+				if (req.user.id === unlockJSON.editorId) {
+					unlockName = {
+						name: unlockJSON.achievement.name
+					};
+				}
+				return unlockName;
+			}
+			catch (err) {
+				return log.debug(err);
+			}
+		});
+
+		alertPromise = Promise.all(promiseList);
+	}
+	else {
+		alertPromise = new Promise(resolve => resolve(false));
+	}
+
+	return (async () => {
```

Comment:
I don't think you need to `return` anything anymore.
Previously we were returning a promise chain, but an `async` function always returns a Promise so that's already taken care of.